internal/providers/gce/gce.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The gce provider fetches a remote configuration from the gce user-data
+// metadata service URL.
+
+package gce
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/coreos/ignition/config"
+	"github.com/coreos/ignition/config/types"
+	"github.com/coreos/ignition/internal/log"
+	"github.com/coreos/ignition/internal/providers"
+	putil "github.com/coreos/ignition/internal/providers/util"
+	"github.com/coreos/ignition/internal/util"
+)
+
+const (
+	initialBackoff = 100 * time.Millisecond
+	maxBackoff     = 30 * time.Second
+	baseUrl        = "http://metadata.google.internal/computeMetadata/v1/instance/"
+	userdataUrl    = baseUrl + "attributes/user-data"
+	sshKeysUrl     = baseUrl + "attributes/ssh-keys"
```

Comment:
The old script reads sshKeys for both project and instance metadata. Looks like ssh-keys is newer and used for instance metadata but not project. Also they seem to be generally fairly complicated and weird these days: https://cloud.google.com/compute/docs/instances/adding-removing-ssh-keys#metadatavalues

the current script: https://github.com/coreos/coreos-overlay/blob/master/coreos-base/oem-gce/files/gce-ssh-key